pkg/types/azure/platform.go

```diff
@@ -28,6 +28,12 @@ type Platform struct {
 
 	// ComputeSubnet specifies an existing subnet for use by compute nodes
 	ComputeSubnet string `json:"computeSubnet,omitempty"`
+	
+	// ResourceGroupName specifies an existing resource group to deploy to
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
```

Comment:
This is not acceptable for IPI installations. this increases the burden on the operators and installer as we needs another way to cleanup.

Reply:
In the case of Azure, this is really just a single action of deleting the resource group. This will delete everything else (except the DNS entries).
If it is desired to be able to be able to offer cleanup, we can pass the ResourceGroupName to the metadata.json file, and then in the destroy action check for this value. If the value is not there fallback to the naming assumption.